src/FSharp.Control.TaskSeq/TaskSeq.fsi

```diff
@@ -871,6 +871,62 @@ type TaskSeq =
+    /// elements. The first element where the predicate returns <see cref="false" /> is returned, which means that this
+    /// function will skip 0 or more elements (see also <see cref="TaskSeq.skipWhileInclusive" />).
+    /// If <paramref name="predicate" /> is asynchronous, consider using <see cref="TaskSeq.skipWhileAsync" />.
+    /// </summary>
+    ///
+    /// <param name="predicate">A function that evaluates to false when no more items should be skipped.</param>
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
+    static member skipWhile: predicate: ('T -> bool) -> source: TaskSeq<'T> -> TaskSeq<'T>
+
+    /// <summary>
+    /// Returns a task sequence that, when iterated, skips elements of the underlying sequence while the
+    /// given asynchronous function <paramref name="predicate" /> returns <see cref="true" />, and then yields the
+    /// remaining elements. The first element where the predicate returns <see cref="false" /> is returned, which
+    /// means that this function will skip 0 or more elements (see also <see cref="TaskSeq.skipWhileInclusive" />).
+    /// If <paramref name="predicate" /> is synchronous, consider using <see cref="TaskSeq.skipWhile" />.
+    /// </summary>
+    ///
+    /// <param name="predicate">An asynchronous function that evaluates to false when no more items should be skipped.</param>
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
+    static member skipWhileAsync: predicate: ('T -> #Task<bool>) -> source: TaskSeq<'T> -> TaskSeq<'T>
+
+    /// <summary>
+    /// Returns a task sequence that, when iterated, skips elements of the underlying sequence until the given
+    /// function <paramref name="predicate" /> returns <see cref="false" />, also skips that element
+    /// and then yields the remaining elements (see also <see cref="TaskSeq.skipWhile" />). This function skips
+    /// at least one element of a non-empty sequence, or returns the empty task sequence if the input is empty.
+    /// If <paramref name="predicate" /> is asynchronous, consider using <see cref="TaskSeq.skipWhileInclusiveAsync" />.
+    /// </summary>`
+    ///
+    /// <param name="predicate">A function that evaluates to false when no more items should be skipped.</param>
+    /// <param name="source">The input task sequence.</param>
+    /// <returns>The resulting task sequence.</returns>
+    /// <exception cref="T:ArgumentNullException">Thrown when the input task sequence is null.</exception>
+    static member skipWhileInclusive: predicate: ('T -> bool) -> source: TaskSeq<'T> -> TaskSeq<'T>
```

Comment:
🤔 I'm back here having tried to read the logic of the impl wondering
- wow I barely read the xmldoc
- should the other one be called exclusive or is should we flip and call it until

Will come back to it later - perhaps after reading our to/fro about why until didnt make sense

Reply:
> should the other one be called exclusive or is should we flip and call it until

@bartelink No. Simple reason, F# Core has `skipWhile` already, so we should have that one regardless. And we already have `takeWhileInclusive`, where `inclusive` applies to what we `take`. 

So, it makes sense that here `inclusive` applies to what we `skip`.

I went back and forth a bit while implementing it, and having it inverse lead to all kinds of other confusions. Now the semantics are equal to the other set of functions we have, with the only difference being "take some items" vs "skip some items".

Reply:
the main reason for not going for `until` was similar: we MUST implement `takeWhile` because it exists in F# Core and we want parity and ease of adoption for users. 

Which means that, if we would go for `until`, we would have to implement both `while` and `until`. I had done that, as you may recall, but your main argument, an explosion of surface area for little gain, was convincing enough for me to drop that PR entirely.

Reply:
skipWhile is clear enough and if we have a `where`, we can and should have a `skipWhile`; no debate
Its wth skipWhileInclusive might mean that's wrecking my head

Reply:
So.... without looking...
- skipWhile skips while the condition holds
- takeWhile takes while the condition holds
- takeWhileInclusive takes while the condition holds, _but also the first one that fails_
- skipWhileInclusive skips while the condition holds but ... the `inclusive` is not telling me intuitively whether it should include the second last one if there was one or also skip another or something

(deleted self-talk about Until having some benefits - no, it doesnt)

But I still don't feel right about `skipWhileInclusive` as I can't think of a use case that I can try names in the context of

As I said, I need to step away and think!

Reply:
WAIT WAIT

The truth table in the overview makes it clear

But the fact you had to do one might be revealing....

Amirite that  `>> TaskSeq.tail` / `>> TaskSeq.drop 1` would achieve the same semantic?

Do we have a use case / is it implemented somewhere else?

Reply:
> Amirite that >> TaskSeq.tail / >> TaskSeq.drop 1 would achieve the same semantic?

It is not that simple. `tail` would raise on empty. Though `drop 1` would be safe, I guess. But it depends on how far the `skipWhile` has gotten. Also, the same can be said for `takeWhileInclusive`, which is just an extra `TaskSeq.head`.

I feel like if we support `takeWhileInclusive`, we should support `skipWhileInclusive` for parity. The same has been done in many functions in both `Seq` and `AsyncSeq` space.

I needed the same truth table for `takeWhileInclusive`, I just never committed it. While working with the functions once you know that they are stable, is much easier. A simple call in FSI shows you what it does.

It's how I tested the edge cases that I wanted to support, by checking `Seq.take/takeWhile/skip/skipWhile` etc.

I'm not dismissing your point, but I find parity and orthogonality very important. Btw, as a comparison, likewise `Seq.map2` is also just `Seq.zip >> Seq.map`, and `Seq.concat` is just `Seq.collect id`, yet they do all exist.

Reply:
PS, if you take this from you:
![image](https://github.com/fsprojects/FSharp.Control.TaskSeq/assets/16015770/dfbe0cc5-0dd5-4e3d-b9bb-012ebbcf4139)

And then I fix it like this:
* skipWhile skips while the condition holds
* takeWhile takes while the condition holds
* takeWhileInclusive takes while the condition holds, but also takes the first one that fails
* skipWhileInclusive skips while the condition holds, but also skips the first one that fails

you can clearly see how they pair together.

Reply:
>It is not that simple. tail would raise on empty. Though drop 1 would be safe, I guess. But it depends on how far the skipWhile has gotten. Also, the same can be said for takeWhileInclusive, which is just an extra TaskSeq.head.

No, the reason for the inclusive is that we want one into the dropped bit - head is no use for that as you'd need to count what you saw, re-iterate skipping that many items etc ?

Reply:
I do get the completeness/having the duals/inverses argument 

But in this case there are many problems:
- Seq does not have it
- The name is wierd
- I honestly cant think of an example concrete usage
- Its hard to explain a perfectly sane SkipWhile as being an opposite of something you cant picture

So I'm still not convinced :(

Reply:
But

>skipWhileInclusive skips while the condition holds, but also skips the first one that fails

this seems a clear explanation, and the table looks logical

but I still don't know what I use it for

Reply:
Next proposal:
>skipWhile skips while the condition holds, but stops before the item that fails

thats overcomplicating matters compared to just saying it skips while the condition holds [, then returns the rest]

The only reason to talk in that way is if you have some other thing to go along with it...

>takeWhile takes while the condition holds, but stops before the item that fails

not sure the ", but stops before the item that fails" adds anything
but, I can see the before/after symmetry
but it feels contrived....

>takeWhileInclusive takes while the condition holds, but stops after the item that fails

while the impl does these things and it is a good way of explaining to someone thinking of the mechanism, its not as good as "like takeWhile but also yields the final item (that failed)"

>skipWhileInclusive skips while the condition holds, but stops after the item that fails

again I cant picture what I'd use it for

So I'm saying I like the first set of explanations more than the last one

But still dont see the what I'd use it for, and there is a tax in having stuff to trawl through and/or rule out if you are never going to use it

----

I get the collect id vs map stuff

But big diff between takeWhileWhileInclusive is that you can't roll your own as a one liner whereas trimming off one more after a skipWhile is relatively trivial.

The ultimatle reason I'm being pedantic is
a) no prior art
b) cant imagine using it
c) it makes it harder to explain the simple normal one

(afk for a bit, sorry for not having time to write it shorter)